test/src/unit/modules/Project.test.js

```diff
@@ -116,32 +202,536 @@ describe('Project.js', () => {
+        });
+        it('Fails to get the metrics of an invalid type', () => {
+            const project = new Project({ name: 'dummy' }, global.codewind.CODEWIND_WORKSPACE);
+            project.loadTestPath = global.codewind.CODEWIND_WORKSPACE;
+            return project.getMetrics('invalidType')
+                .should.be.eventually.rejectedWith('invalidType is not a valid metric type. Valid metric types are\n[cpu,gc,memory,http]')
+                .and.be.an.instanceOf(ProjectError)
+                .and.have.property('code', 'INVALID_METRIC_TYPE');
+        });
+        it('Gets an empty array of metrics as none exist in the loadTestPath', async() => {
+            const project = new Project({ name: 'dummy' }, global.codewind.CODEWIND_WORKSPACE);
+            project.loadTestPath = global.codewind.CODEWIND_WORKSPACE;
+            const cpuMetrics = await project.getMetrics('cpu');
+            cpuMetrics.should.be.an('array');
+            cpuMetrics.should.be.empty;
+        });
+        describe('Gets metrics using the load tests in the resources directory', () => {
+            const metrics = ['cpu', 'gc', 'memory', 'http'];
+            metrics.forEach(function(type) {
+                it(type, async() => {
+                    const files = await fs.readdir(loadTestResources);
+                    const noLoadTests = files.length;
+                    const project = new Project({ name: 'dummy' }, global.codewind.CODEWIND_WORKSPACE);
+                    project.loadTestPath = loadTestResources;
+                    const cpuMetrics = await project.getMetrics(type);
+                    cpuMetrics.should.have.length(noLoadTests);
+                    cpuMetrics.forEach(function(results) {
+                        results.should.have.property('value');
+                        results.value.should.have.property('data');
+                    });
+                });
+            });
+        });
+    });
+    describe('getMetricsByTime(timeOfTestRun)', () => {
+        it('Fails to get a metrics file using an invalid time stamp', () => {
+            const project = new Project({ name: 'dummy' }, global.codewind.CODEWIND_WORKSPACE);
+            project.loadTestPath = loadTestResources;
+            return project.getMetricsByTime('123')
+                .should.be.eventually.rejectedWith(`Unable to find metrics for project ${project.projectID}`)
+                .and.be.an.instanceOf(ProjectMetricsError)
+                .and.have.property('code', 'NOT_FOUND');
+        });
+        it('Gets a metrics file using a valid time stamp (String)', async() => {
+            const project = new Project({ name: 'dummy' }, global.codewind.CODEWIND_WORKSPACE);
+            project.loadTestPath = loadTestResources;
+            const metricsFile = '20190326154749';
+            const metricsTypes = await project.getMetricsByTime(metricsFile);
+            metricsTypes.should.have.property('id').and.equal(0);
```

Comment:
this could be array of properties which you loop through, to save some duplication